src/providers/cobol.rs

```diff
@@ -0,0 +1,115 @@
+use std::{path::PathBuf, str::FromStr};
+
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    nix::pkg::Pkg,
+    plan::{
+        phase::{Phase, StartPhase},
+        BuildPlan,
+    },
+};
+use anyhow::Result;
+use path_slash::PathBufExt;
+
+const COBOL_COMPILE_ARGS: &str = "COBOL_COMPILE_ARGS";
+const COBOL_APP_NAME: &str = "COBOL_APP_NAME";
+const DEFAULT_COBOL_COMPILE_ARGS: &str = "-x -o";
+
+pub struct CobolProvider {}
+
+impl Provider for CobolProvider {
+    fn name(&self) -> &str {
+        "cobol"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.has_match("*.cbl"))
+    }
+
+    fn get_build_plan(
+        &self,
+        app: &App,
+        environment: &Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::plan::BuildPlan>> {
+        let setup = Phase::setup(Some(vec![Pkg::new("gnu-cobol"), Pkg::new("gcc")]));
+
+        let compile_args = environment
+            .get_config_variable(COBOL_COMPILE_ARGS)
+            .unwrap_or_else(|| DEFAULT_COBOL_COMPILE_ARGS.to_string());
+
+        let app_path = CobolProvider::get_app_path(self, app, environment);
+        let mut file_name = "";
+
+        if let Some(file_stem) = app_path.file_stem() {
+            if let Some(unwrapped_file_name) = file_stem.to_str() {
+                file_name = unwrapped_file_name;
+            }
+        }
```

Comment:
Nit: This can be cleaned up a bit by using `and_then` and `unwrap_or_default`. Something like

```rs
 let file_name = app_path
  .file_stem()
  .and_then(|stem| stem.to_str())
  .unwrap_or_default();
```

A few of the other _triangles_ can also be de-trianglified using a similar approach

Reply:
perfect! Ive removed the triangles using this method, and refactored the code a tad.